lib/models/room-state.js

```diff
@@ -225,6 +225,52 @@ RoomState.prototype.getUserIdsWithDisplayName = function(displayName) {
 };
 
 /**
+ * Returns true if the given user ID has permission to send a state
+ * event of type `stateEventType` into this room.
+ * @param {string} type The type of state events to test
+ * @param {string} userId The user ID of the user to test permission for
+ * @return {boolean} true if the given user ID should be permitted to send
+ *                        the given type of state event into this room,
+ *                        according to the room's state.
+ */
+RoomState.prototype.maySendStateEvent = function(stateEventType, userId) {
+    if (!this.getMember(userId)) { return false; }
```

Comment:
This only tests for presence of `m.room.member` events. This will return a `RoomMember` with `membership = leave` if they leave the room.